perfkitbenchmarker/providers/azure/azure_kubernetes_service.py

```
@@ -392,6 +392,89 @@ def GetNodePoolNames(self) -> list[str]:
 
+class AksAutomaticCluster(AksCluster):
+  """Class representing an AKS Automatic cluster, which has managed node pools."""
+
+  CLOUD = provider_info.AZURE
+  CLUSTER_TYPE = 'Auto'
+
+  def __init__(self, spec):
+    """Initializes the cluster."""
+    super().__init__(spec)
+
+  def _Create(self):
+    """Creates the Automatic AKS cluster with tags."""
+    tags_dict = util.GetResourceTags(self.resource_group.timeout_minutes)
+    tags_list = [f'{k}={v}' for k, v in tags_dict.items()]
+    cmd = [
+        azure.AZURE_PATH,
+        'aks',
+        'create',
+        '--name',
+        self.name,
+        '--location',
+        self.region,
+        '--ssh-key-value',
+        vm_util.GetPublicKeyPath(),
+        '--resource-group',
+        self.resource_group.name,
+        '--sku',
+        'automatic',
+        '--tags',
+    ] + tags_list
+    vm_util.Retry(timeout=300)(vm_util.IssueCommand)(
+        cmd,
+        # Half hour timeout on creating the cluster.
+        timeout=1800,
+    )
+
+  def _IsReady(self):
+    """Returns True if the cluster is ready."""
+    # Check provisioning state
+    show_cmd = [
+        azure.AZURE_PATH,
+        'aks',
+        'show',
+        '--name',
+        self.name,
+    ] + self.resource_group.args
+    stdout, _, _ = vm_util.IssueCommand(show_cmd, raise_on_failure=False)
+    try:
+        cluster = json.loads(stdout)
+        if cluster.get('provisioningState') != 'Succeeded':
+            return False
+    except Exception:
+        return False
+
+    vm_util.IssueCommand(
+        [
+            azure.AZURE_PATH,
+            'aks',
+            'get-credentials',
+            '--name',
+            self.name,
+            '--file',
+            FLAGS.kubeconfig,
+        ]
+        + self.resource_group.args
+    )
+    version_cmd = [FLAGS.kubectl, '--kubeconfig', FLAGS.kubeconfig, 'version']
+    _, _, retcode = vm_util.IssueCommand(version_cmd, raise_on_failure=False)
+    if retcode:
+      return False
```

Likewise we are swallowing a lot of errors in _IsReady.. but here it makes sense as we need to wait until these things are indeed ready & expect some errors before then. 

You still may encounter real vs expected errors here, but it mostly matters for debugging (hiding error as timeout rather than real error, which can still be found after further investigation) & faster timeouts. 

Please add some documentation of the error messages encountered. There should be an azure_kubernetes_service_test.py you could add documentation as code unit tests. Otherwise just in comments.

---

This handling checks the cluster's `provisioningState` and waits until it becomes `Succeeded`.

However, in certain cases, the cluster may not reach the `"provisioningState": "Succeeded"` (`"provisioningState": "Updating"`), which could result in failed subsequent deployment attempts.

Please let me know how you would like to handle this.